File: docs_old/datatools/codesnippet/VisualBasic/how-to--save-dataset-changes-to-a-database_3.vb
    Private Sub UpdateDB()
        Dim deletedChildRecords As NorthwindDataSet.OrdersDataTable =
            CType(NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Deleted), NorthwindDataSet.OrdersDataTable)

        Dim newChildRecords As NorthwindDataSet.OrdersDataTable =
            CType(NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Added), NorthwindDataSet.OrdersDataTable)

        Dim modifiedChildRecords As NorthwindDataSet.OrdersDataTable =
            CType(NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Modified), NorthwindDataSet.OrdersDataTable)

        Try
            If deletedChildRecords IsNot Nothing Then
                OrdersTableAdapter.Update(deletedChildRecords)
            End If

            CustomersTableAdapter.Update(NorthwindDataSet.Customers)

            If newChildRecords IsNot Nothing Then
                OrdersTableAdapter.Update(newChildRecords)
            End If

            If modifiedChildRecords IsNot Nothing Then
                OrdersTableAdapter.Update(modifiedChildRecords)
            End If

            NorthwindDataSet.AcceptChanges()

        Catch ex As Exception
            MessageBox.Show("An error occurred during the update process")
            ' Add code to handle error here.

        Finally
            If deletedChildRecords IsNot Nothing Then
                deletedChildRecords.Dispose()
            End If

            If newChildRecords IsNot Nothing Then
                newChildRecords.Dispose()
            End If

            If modifiedChildRecords IsNot Nothing Then
                modifiedChildRecords.Dispose()
            End If

        End Try
    End Sub

File: docs/data-tools/codesnippet/CSharp/how-to-save-dataset-changes-to-a-database_3.cs
        void UpdateDB()
        {
            NorthwindDataSet.OrdersDataTable deletedChildRecords = 
                (NorthwindDataSet.OrdersDataTable)northwindDataSet.Orders.GetChanges(DataRowState.Deleted);

            NorthwindDataSet.OrdersDataTable newChildRecords = 
                (NorthwindDataSet.OrdersDataTable)northwindDataSet.Orders.GetChanges(DataRowState.Added);

            NorthwindDataSet.OrdersDataTable modifiedChildRecords = 
                (NorthwindDataSet.OrdersDataTable)northwindDataSet.Orders.GetChanges(DataRowState.Modified);

            try
            {
                if (deletedChildRecords != null)
                {
                    ordersTableAdapter.Update(deletedChildRecords);
                }

                customersTableAdapter.Update(northwindDataSet.Customers);

                if (newChildRecords != null)
                {
                    ordersTableAdapter.Update(newChildRecords);
                }

                if (modifiedChildRecords != null)
                {
                    ordersTableAdapter.Update(modifiedChildRecords);
                }

                northwindDataSet.AcceptChanges();
            }

            catch (Exception ex)
            {
                MessageBox.Show("An error occurred during the update process");
                // Add code to handle error here.
            }

            finally
            {
                if (deletedChildRecords != null)
                {
                    deletedChildRecords.Dispose();
                }
                if (newChildRecords != null)
                {
                    newChildRecords.Dispose();
                }
                if (modifiedChildRecords != null)
                {
                    modifiedChildRecords.Dispose();
                }
            }
        }